### DISCUSSION TITLE
Problema com boleto CEF com Código do Beneficiário com 7 dígitos

### DISCUSSION THREAD

**Discussion by alessandrobrunolima** [braully/bopepo#18] [Q&A] [answered]
Bom dia, alguém teve problema com boleto da CEF no modelo SIGCB com código de beneficiário com 7 dígitos?

**Comment by braully** [comment]
Não gero um boleto na caixa a muitos anos, não tive problemas na época. Qual o problema você está tendo? Verificar a discussão na lib original, talvez consiga uma solução e pode mandar para cá que eu faço o Merge.


Verificar solução em: https://github.com/jrimum/bopepo/issues/18

**Reply by alessandrobrunolima** [reply]
Boa tarde Braully, o meu problema acontece ao tentar gerar um boleto da caixa com o código do beneficiário com 7 dígitos, uma vez que o código do beneficiário com 6 dígitos é informado na entidade NumeroDaConta.codigoDaConta. Mas não consigo informar nesse campo o código do beneficiário com 7 dígitos, então comecei a passar as 6 primeiras posições no codigoDaConta e a ultima posição no digitoDaConta, mas mesmo assim o boleto não passa na homologação da CEF.

**Reply by braully** [reply]
O dígito verificador da conta deve ser informado em campo separado,  veja exemplo https://github.com/braully/bopepo/blob/master/src/test/java/org/jrimum/bopepo/campolivre/TestCLCaixaEconomicaFederalSIGCB.java

**Reply by alessandrobrunolima** [reply]
Acho que encontrei o problema, mesmo eu passando o digito verificador a classe CLCaixaEconomicaFederalSIGCB não utiliza esse digito para fazer a montagem do campo livre. Acredito que terei q criar minha própria classe de Campo Livre. 
Caso tenha sucesso eu te informo. Muito obrigado pela atenção e ajuda.

**Comment by braully** [comment]
Ok, caso consiga, faça um fork, altere, teste e me mande um pull request.

**Reply by alessandrobrunolima** [reply, marked as answer]
Bom dia, com a correria aqui acabei atrasando a criação do fork e do pull request, mas acabei de criar o pull request.

**Reply by braully** [reply]
Pull request aceito. Logo libero a nova versão no maven com sua alteração. Obrigado pela contribuição 😃
